pkg/cache/cache.go

```diff
@@ -147,31 +134,36 @@ type ViewOptions struct {
 	DefaultTransform toolscache.TransformFunc
 
 	// ByObject restricts the cache's ListWatch to the desired fields per GVK at the specified object.
-	ByObject ViewByObject
+	ByObject map[client.Object]ByObject
+
+	// UnsafeDisableDeepCopy indicates not to deep copy objects during get or
+	// list objects for EVERY object.
+	// Be very careful with this, when enabled you must DeepCopy any object before mutating it,
+	// otherwise you will mutate the object in the cache.
```

Comment:
Nit: mention that this can be overrriden through `ByObject` per GVK?

Reply:
fixed, thanks